samples/cut-geometry/src/main/java/com/esri/arcgismaps/sample/cutgeometry/components/CutGeometryViewModel.kt

```diff
@@ -0,0 +1,215 @@
+    private val _isResetButtonEnabled = MutableStateFlow(false)
+    val isResetButtonEnabled = _isResetButtonEnabled.asStateFlow()
+
+    private val _isCutButtonEnabled = MutableStateFlow(false)
+    val isCutButtonEnabled = _isCutButtonEnabled.asStateFlow()
+
+    // create a graphic overlay
+    val graphicsOverlay = GraphicsOverlay()
+
+    // create a message dialog view model for handling error messages
+    val messageDialogVM = MessageDialogViewModel()
+
+    init {
+        viewModelScope.launch {
+            arcGISMap.load().onFailure { error ->
+                messageDialogVM.showMessageDialog(
+                    "Failed to load map",
+                    error.message.toString()
+                )
+            }.onSuccess {
+                graphicsOverlay.graphics.add(polylineGraphic)
+                graphicsOverlay.graphics.add(polygonGraphic)
+                polygonGraphic.geometry?.let { polygonToCut ->
+                    mapViewProxy.setViewpoint(Viewpoint(polygonToCut))
+                }
+                _isCutButtonEnabled.value = true
+            }
+        }
+    }
+
+    /**
+     * Clear the current graphics, then re-add the graphics for Lake Superior and the cut polyline
+     * */
+    fun resetGeometry() {
+        graphicsOverlay.graphics.clear()
+        graphicsOverlay.graphics.add(polylineGraphic)
+        graphicsOverlay.graphics.add(polygonGraphic)
+        polygonGraphic.geometry?.let { polygonToCut ->
+            mapViewProxy.setViewpoint(Viewpoint(polygonToCut))
+        }
+
+        _isResetButtonEnabled.value = false
+        _isCutButtonEnabled.value = true
+    }
+
+    /**
+     * Cut the Lake Superior graphic into a US side and Canada side using the cut polyline
+     * and then add the resulting graphics to the graphics overlay
+     */
+    fun cutGeometry() {
+        polygonGraphic.geometry?.let { graphicGeometry ->
+            val parts = GeometryEngine.tryCut(
+                graphicGeometry,
+                polylineGraphic.geometry as Polyline
+            )
+
+            // create graphics for the US and Canada sides
+            val canadaSide = Graphic(
+                parts[0], SimpleFillSymbol(
+                    SimpleFillSymbolStyle.BackwardDiagonal,
+                    Color.green, SimpleLineSymbol(SimpleLineSymbolStyle.Null, Color.blue, 0F)
+                )
+            )
+            val usSide = Graphic(
+                parts[1], SimpleFillSymbol(
+                    SimpleFillSymbolStyle.ForwardDiagonal,
+                    Color.yellow, SimpleLineSymbol(SimpleLineSymbolStyle.Null, Color.blue, 0F)
+                )
+            )
+            // add the graphics to the graphics overlay
+            graphicsOverlay.graphics.addAll(listOf(canadaSide, usSide))
+
+            // update button state
+            _isCutButtonEnabled.value = false
+            _isResetButtonEnabled.value = true
+        }
+    }
+
+    /**
+     * Companion for a nicer color
+     */
```

Comment:
Should probably be more descriptive. Feel free to rewrite:
```suggestion
    /**
     * Define a blue color for polygon boundary
     */
```